File: bubbleSort.py
def bubbleSort(arr) :
  n = len(arr)
  
  for i in range(n):
    for j in range(0, n-i-1):
      if arr[j] > arr[j+1]:
        arr[j], arr[j+1] = arr[j+1], arr[j]
  return arr

arr = [99, 12, 46, 235, 1, 3]
k = 3
bubbleSort(arr)
print(arr)
# print greatest 3 numbers
print(arr[k:])

def bubbleSortReverse(arr) :
  n = len(arr)
  
  for i in range(n):
    for j in range(0, n-i-1):
      if arr[j] < arr[j+1]:
        arr[j], arr[j+1] = arr[j+1], arr[j]
  return arr

arr = [99, 12, 46, 235, 1, 3]

bubbleSortReverse(arr)
print(arr)
# print greatest 3 numbers
print(arr[:k])

File: bubbleSort.js
//JavaScript way to bubble sort
let arr = [99, 325, 23, 12, 687, 1, 6];

const bubbleSortReverse = (arr) => {
  // define n as length of array so we can use its value later in the codes for loops
  let n = arr.length;
  // this first for loop allows us to loop over each element j n-i-1 amount of times. This will taper down to zero by the time i gets to the last element in the list
  for (let i = 0; i < n; i++){
    for (let j = 0; j < n-i-1; j++){
      //if arr at j is less and array at j+1, then swap them.
      if(arr[j] < arr[j+1]){
        //set arr at j to temp to save so we can then set arr at j+1 to arr at j later on.
        temp = arr[j];
        arr[j] = arr[j+1]
        arr[j+1] = temp
      }
    }
  }
  return arr;
}

const bubbleSort = (arr) => {
  let n = arr.length;
  for (let i = 0; i < n; i++){
    for (let j = 0; j < n-i-1; j++){
      if(arr[j] > arr[j+1]){
        temp = arr[j];
        arr[j] = arr[j+1]
        arr[j+1] = temp
      }
    }
  }
  return arr;
}

bubbleSort(arr);
console.log(arr);


bubbleSortReverse(arr);
console.log(arr);



$("#1").attr(data-crystalValue, crystal1)
$("#2").attr(data-crystalValue, crystal2)
$("#3").attr(data-crystalValue, crystal3)
$("#4").attr(data-crystalValue, crystal4)
